gcc/rust/typecheck/rust-hir-type-check-expr.h

```diff
@@ -702,14 +706,25 @@ class TypeCheckExpr : public TypeCheckBase
       return true;
     });
 
-    infered_array_elems = types[0];
-    for (size_t i = 1; i < types.size (); i++)
+    // Create an infer type and register it in TypeCheckContext
+    // TODO: This is very leaky. Need a refactor of some sort.
+    auto crateNum = Analysis::Mappings::get ()->get_current_crate ();
+    auto nodeId = UNKNOWN_NODEID;
+    auto hirId = mappings->get_next_hir_id ();
+    auto localDefId = UNKNOWN_LOCAL_DEFID;
+    auto mappings = Analysis::NodeMapping (crateNum, nodeId, hirId, localDefId);
+    infered_array_elems
+      = new TyTy::InferType (hirId, TyTy::InferType::InferTypeKind::GENERAL);
```

Comment:
I have a wrapper over this that you can use:

https://github.com/Rust-GCC/gccrs/blob/e997db044394dd7ae751f52c209320a0097218be/gcc/rust/typecheck/rust-tyty.cc#L51

You should be able to call:

```c++
infered_array_elems = TyTy::TyVar::get_implict_infer_var ();
```